onnxruntime/core/optimizer/graph_transformer_mgr.h

```diff
@@ -4,31 +4,63 @@
 #pragma once
 
 #include "core/optimizer/graph_transformer.h"
-
+#include "core/optimizer/graph_transformer_factory.h"
+using namespace ::onnxruntime::common;
 namespace onnxruntime {
-// Manages a list of graph transformers. It is initialized with a list of graph
+
+// Manages graph transformation for a inference session. It is initialized with a list of pre-defined graph
 // transformers. Each inference session can further register additional ones.
 class GraphTransformerManager {
  public:
-  explicit GraphTransformerManager(unsigned steps) noexcept : steps_(steps) {
-    // TODO: Register default transformers.
+  explicit GraphTransformerManager(unsigned steps, unsigned int level) : steps_(steps), factory_{std::make_unique<GraphTransformerFactory>()} {
+    ORT_ENFORCE(ValidateOptimizationLevel(level).IsOK(),
+                "Unsupported optimization level specified");
+
```

Comment:
Error message should ideally say what the valid level/s are and not just that the input was invalid. That way the user knows how to fix the issue.